sdk/tables/azure-tables/src/serialization.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { base64Encode, base64Decode } from "./utils/bufferSerializer";
+
+const Edm = {
+  Binary: "Edm.Binary",
+  Boolean: "Edm.Boolean",
+  DateTime: "Edm.DateTime",
+  Double: "Edm.Double",
+  Guid: "Edm.Guid",
+  Int32: "Edm.Int32",
+  Int64: "Edm.Int64",
+  String: "Edm.String"
+};
+
+type supportedTypes = boolean | string | number | Date | Uint8Array;
+
+type serializedType = {
+  value: supportedTypes;
+  type?: string;
+};
+
+function serializePrimitive(value: any): serializedType {
+  const serializedValue: serializedType = { value };
+  if (typeof value === "boolean" || typeof value === "string" || typeof value === "number") {
+    serializedValue.value = value;
+  } else if (value instanceof Date) {
+    serializedValue.value = value;
+    serializedValue.type = Edm.DateTime;
+  } else if (value instanceof Uint8Array) {
+    serializedValue.value = base64Encode(value);
+    serializedValue.type = Edm.Binary;
+  } else {
+    throw new Error(`Unknown EDM type ${typeof value}`);
+  }
+
+  return serializedValue;
+}
+
+function serializeObject(obj: { value: any; type: string }): serializedType {
+  const serializedValue: serializedType = { value: obj.value };
+  if (
+    obj.type === "Boolean" ||
+    obj.type === "DateTime" ||
+    obj.type === "Double" ||
+    obj.type === "Guid" ||
+    obj.type === "Int32" ||
+    obj.type === "Int64" ||
+    obj.type === "String"
+  ) {
+    serializedValue.value = obj.value;
+    serializedValue.type = Edm[obj.type];
+  } else if (obj.type === "Binary") {
+    serializedValue.value = base64Encode(obj.value);
+    serializedValue.type = Edm.Binary;
+  } else {
+    throw new Error(`Unknown EDM type ${typeof obj.value}`);
+  }
+
+  return serializedValue;
+}
+
+function getSerializedValue(value: any): serializedType {
+  if (typeof value === "object" && "value" in value && "type" in value) {
```

Comment:
since we're using `in` here we are checking up the prototype chain - is that expected?